core/config.go

```diff
@@ -333,6 +335,10 @@ func (c *ChainConfig) GetFeature(num *big.Int, id string) (*ForkFeature, *Fork,
 }
 
 func (c *ChainConfig) HeaderCheck(h *types.Header) error {
+	if (h == &types.Header{}) {
+		return ErrHashEmpty
```

Comment:
I figured it was better to return a specific error because "ignoring" (return a non-error for) an empty header seemed less "extensible" than being more explicit about what's expected. 

Aside from `eth/handler.go`, `HeaderCheck` is also called here at `InsertChain` - https://github.com/ethereumproject/go-ethereum/blob/master/core/blockchain.go#L850.

It may be that an empty header _should_ only happen in tests, but I thought I would rather be certain to receive a descriptive error in case it happens unexpectedly.

Does that make sense?